RocketChat/Rocket.Chat.ReactNative
# Chore: Update WatermelonDB to 0.23.0

## Proposed changes

Updates @nozbe/watermelondb to 0.23.0.
After this PR lands, we can make use of JSI.

## Issue(s)	

https://app.clickup.com/t/5n4v60

## How to test or reproduce

- Test general features that involves local database: channels, messages, close and resume the app, etc
- Notifications should be working properly
  - Test on open server
  - Test on https://mobile.rocket.chat, because push notification data privacy is enabled there

## Screenshots

## Types of changes




## Checklist



## Further comments
